src/main/java/com/crud/library/controller/RentalController.java

```diff
@@ -0,0 +1,35 @@
+package com.crud.library.controller;
+
+import com.crud.library.dbService.RentalService;
+import com.crud.library.domain.RentalDto;
+import com.crud.library.exception.RentalAlreadyExistsException;
+import com.crud.library.exception.RentalNotFoundException;
+import com.crud.library.exception.UserNotExistsException;
+import com.crud.library.exception.VolumeNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/library/rental")
+@CrossOrigin(origins = "*")
+@RequiredArgsConstructor
+public class RentalController {
+    private final RentalService rentalService;
+
+    @PostMapping (value = "create")
+    public RentalDto createRental(@RequestParam long volumeId, @RequestParam long userId) throws RentalAlreadyExistsException, UserNotExistsException, VolumeNotFoundException {
+        return rentalService.createRental(volumeId, userId);
+    }
+
+    @GetMapping("get")
+    public List<RentalDto> getRentals() {
+        return rentalService.getAllRentals();
+    }
+
+    @DeleteMapping(value = "delete")
+    public void deleteRental(@RequestParam long volumeId) throws RentalNotFoundException, VolumeNotFoundException {
+        rentalService.deleteRental(volumeId);
+    }
+}
```

Comment:
generalna uwaga do wszystkich tego typu miejsc:
1) Urle endpointów składają się z rzeczowników: np. /library/rental (bez get, create, delete), w tym kontrolerze wystarczy jeden url na wszystkie trzy endpointy - odróżnia je metoda HTTP (POST< GET, DELETE)
2) Jeśli podajemy id czegoś, to podajemy to jako @Pathariable a nie @RequestParam
3) po co dodawać suffix Rental skoro jesteśmy w klasie RENTALController?
4) Metoda POST i PUT powinny mieć @RequestBody z przekazywanym DTO

To powinno wyglądać tak:

@PostMapping 
    public RentalDto create(@RequestBody rentalDto) throws RentalAlreadyExistsException, UserNotExistsException, VolumeNotFoundException {
        return rentalService.create(rentalDto.getVolumeId(), rentalDto.getUserId());
    }

    @GetMapping
    public List<RentalDto> getAll() {
        return rentalService.getAll();
    }

    @DeleteMapping({volumeId})
    public void delete(@PathVariable("volumeId") long volumeId) throws RentalNotFoundException, VolumeNotFoundException {
        rentalService.delete(volumeId);
    }

A gdzie metoda uzyskania rentala po id?

@GetMapping(/{rentalId})
    public List<RentalDto> get(@PathVariable("rentalId") long rentalId) {
        return rentalService.get(rentalId);
    }